GoldRush.js

```diff
@@ -0,0 +1,123 @@
+            switch (direction) {
+                case 'up':
+                    player.coords.y--
+                    this.goldRush.alter(player.coords.x, player.coords.y, playerNumber)
+                    this.goldRush.alter(player.coords.x, player.coords.y + 1, ".")
+                    break
+                case 'down':
+                    player.coords.y++
+                    this.goldRush.alter(player.coords.x, player.coords.y, playerNumber)
+                    this.goldRush.alter(player.coords.x, player.coords.y - 1, ".")
+                    break
+                case 'left':
+                    player.coords.x--
+                    this.goldRush.alter(player.coords.x, player.coords.y, playerNumber)
+                    this.goldRush.alter(player.coords.x + 1, player.coords.y, ".")
+                    break
+                case 'right':
+                    player.coords.x++
+                    this.goldRush.alter(player.coords.x, player.coords.y, playerNumber)
+                    this.goldRush.alter(player.coords.x - 1, player.coords.y, ".")
+                    break
+                default:
+                    console.log(`Sorry, we are out of ${direction}.`)
+            }
+        }
+
+        if(this.collectCoin(playerNumber)){
+            this.incrementScore(playerNumber)
+        }
+        this.checkVictory()
+    }
+
+    placeCoins = function () {
+        let counter = 0
+        while (counter != Math.floor(this.rowNum*this.colNum/4)) {
+            let randomRow = Math.floor(Math.random() * this.rowNum)
+            let randomColumn = Math.floor(Math.random() * this.colNum)
+            if (this.checkIfFree(randomRow, randomColumn)) {
+                this.goldRush.alter(randomRow, randomColumn, "c")
+                this.coinCoords.push({x: randomRow, y: randomColumn})
+                counter++
+            }
+        }
+    }
+
+    checkIfFree = function (i, j) {
+        return this.goldRush.matrix[j][i] == "."
+    }
+
+    collectCoin = function (playerNumber) {
+        let player = this.determinePlayer(playerNumber)
+        let isExist = this.coinCoords.find(elem => elem.x==player.coords.x && elem.y==player.coords.y)
+        if (isExist) {
+            let coinIndex = this.coinCoords.findIndex(elem => elem.x==player.coords.x && elem.y==player.coords.y)
+            this.coinCoords.splice(coinIndex, 1)
+            return true
+        }
+        return false
+    }
+
+    isAvailableStep = function(x, y, dir){
+        return !((dir=="up"&&y==0) || (dir=="down"&&y==this.colNum-1) || (dir=="left"&&x==0) || (dir=="right"&&x==this.rowNum-1))
+    }
+
+    incrementScore = function (playerNumber) {
+        if (playerNumber === 1){
+            this.score.firstPlayer += 10
+        } else {
+            this.score.secondPlayer += 10
+        }
+    }
+
+    checkVictory = function () {
+        if (!this.coinCoords.length) {
+            let bool = this.score.secondPlayer > this.score.firstPlayer
```

Comment:
You don't need both lines, you can just do:
`let winner = this.score.secondPlayer > this.score.firstPlayer ? 2 : 1`